packages/adminrouter/extra/src/test-harness/modules/generic_test_code/generalised_tests.py

```diff
@@ -305,6 +311,71 @@ def _testdata_to_redirect_testdata(tests_config, node_type):
     return res
 
 
+def _universal_test_if_upstream_headers_are_correct(
+        self,
+        ar_process,
+        valid_user_header,
+        path,
+        jwt_forwarded_test,
+        skip_authcookie_filtering_test,
+        ):
+
+    headers_present = {}
+    headers_absent = []
+
+    if jwt_forwarded_test is True:
+        headers_present.update(valid_user_header)
+    elif jwt_forwarded_test is False:
+        headers_absent.append("Authorization")
+    # jwt_forwarded_test == "skip", do nothing
+
+    generic_upstream_headers_verify_test(
+        ar_process,
+        valid_user_header,
+        path,
+        assert_headers=headers_present,
+        assert_headers_absent=headers_absent,
+        )
+
+    if skip_authcookie_filtering_test:
+        return
+
+    filtered_cookies = ['dcos-acs-info-cookie', 'dcos-acs-auth-cookie']
+    jar = {}
+    # both dcos-* cookies should be removed by AR
+    # the dcos-acs-auth-cookie cookie is simply the DC/OS authentication token:
+    jar['dcos-acs-auth-cookie'] = valid_user_header['Authorization']
+    # the dcos-acs-info-cookie is base64-encoded data for dc/os UI,
+    # currently set to default user and password
```

Comment:
> currently set to default user and password

What does that mean, and where does this come from? :)

The info cookie is a cookie accessible by JavaScript code (as opposed to the auth cookie) and just contains the `uid` of the user that is currently logged in, so that the UI can display it.